File: ChangeColorApp/app/src/main/java/kz/kbtu/changecolorapp/MainActivity.java
package kz.kbtu.changecolorapp;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    private Button btnChangeColor;
    private View v;
    private Random rand;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        rand = new Random();
        btnChangeColor = findViewById(R.id.btn_change_color);
        v = findViewById(R.id.root_view);

        v.setBackgroundColor(Color.RED);
        btnChangeColor.setBackgroundColor(Color.BLUE);

        btnChangeColor = findViewById(R.id.btn_change_color);
        v = findViewById(R.id.root_view);

        v.setBackgroundColor(Color.RED);
        btnChangeColor.setBackgroundColor(Color.BLUE);

        btnChangeColor.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                int colorBtn = ((ColorDrawable)btnChangeColor.getBackground()).getColor();
                int colorView = ((ColorDrawable)v.getBackground()).getColor();
                v.setBackgroundColor(colorBtn);
                btnChangeColor.setBackgroundColor(colorView);
            }
        });
    }

}

File: KotlinTestApp/app/src/main/java/kz/kbtu/kotlintestapp/MainActivity.kt
package kz.kbtu.kotlintestapp

import android.graphics.Color
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import java.util.*

class MainActivity : AppCompatActivity(), View.OnClickListener {



    lateinit var btnClick : Button
    lateinit var rootView : LinearLayout
    var random = Random()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        bindViews()
    }

    private fun bindViews(){
        btnClick = findViewById(R.id.button)
        rootView = findViewById(R.id.layout)
        btnClick.setOnClickListener(this)
    }


    override fun onClick(p0: View?) {
        when(p0!!.id){
            R.id.button -> buttonClicked()
        }
    }

    private fun buttonClicked() {
        rootView.setBackgroundColor(Color.rgb(random.nextInt(255), random.nextInt(255), random.nextInt(255)))
        onClick(null)
    }

    private fun isPrime(a : Int) : Boolean{
        return a % 2 != 0
    }


}
